### PR TITLE
Add response code probabilities option

### PR DESCRIPTION
**Pull request by peteraba**
https://github.com/gernest/apidemic/issues/3

@gernest This PR is based on code from https://github.com/gernest/apidemic/pull/6 so please check that first.

### CODE REVIEW

**Comment by gernest** [comment]
Please rebase on master and resolve the conflicts.

**Comment by peteraba** [comment]
@gernest rebased

File: api_test.go [hunk 1]
```diff
@@ -54,6 +60,32 @@ func TestDynamicEndpointWithPostRequest(t *testing.T) {
...
+
+	w = httptest.NewRecorder()
+	req = jsonRequest("GET", "/api/test", nil)
+
+	s.ServeHTTP(w, req)
+	assert.Equal(t, http.StatusForbidden, w.Code)
+}
+
+func setUp() *mux.Router {
+	store = func() *cache.Cache {
```
Comment (gernest) [inline api_test.go]: No need for a function to initialize the value.  I think this will do

```
	store = cache.New(5*time.Minute, 30*time.Second)
```

**Comment by peteraba** [comment]
@gernest fixed.

**Pull request merged**